src/test/java/io/github/enkarin/bookcrossing/service/UserServiceTest.java

```diff
@@ -0,0 +1,147 @@
+        usersId.clear();
+    }
+
+    @Test
+    void saveUserCorrectUserTest() {
+        final UserDto user = userService.saveUser(TestDataProvider.buildAlex());
+        usersId.add(user.getUserId());
+
+        assertThat(userService.findById(user.getUserId(), 0)).isNotNull();
+    }
+
+    @Test
+    void saveUserNonConfirmedPasswordTest() {
+        assertThatThrownBy(() -> userService.saveUser(TestDataProvider.buildNonConfirmedPasswordUser()))
+                .isInstanceOf(PasswordsDontMatchException.class)
+                .hasMessage("Пароли не совпадают");
+    }
+
+    @Test
+    void saveUserDuplicatedUserTest() {
+        final UserDto user = userService.saveUser(TestDataProvider.buildAlex());
+        usersId.add(user.getUserId());
+
+        assertThatThrownBy(() -> userService.saveUser(TestDataProvider.buildAlex()))
+                .isInstanceOf(LoginFailedException.class)
+                .hasMessage("login: Пользователь с таким логином уже существует");
+    }
+
+    @Test
+    void saveUserDuplicatedEmailTest() {
+        final UserDto user = userService.saveUser(TestDataProvider.buildAlex());
+        usersId.add(user.getUserId());
+
+        assertThatThrownBy(() -> userService.saveUser(TestDataProvider.buildUserWithAlexEmail()))
+                .isInstanceOf(EmailFailedException.class)
+                .hasMessage("email: Пользователь с таким почтовым адресом уже существует");
+    }
+
+    @Test
+    void confirmMailWrongTokenTest() {
+        assertThatThrownBy(() -> userService.confirmMail("wrong token"))
+                .isInstanceOf(TokenNotFoundException.class)
+                .hasMessage("Токен не найден");
+    }
+
+    @Test
+    void findByLoginCorrectTest() {
+        final UserDto user = userService.saveUser(TestDataProvider.buildAlex());
+        usersId.add(user.getUserId());
+
+        assertThat(userService.findByLogin(user.getLogin())).isNotNull();
+    }
+
+    @Test
+    void findByLoginNonExistingUserTest() {
+        assertThatThrownBy(() -> userService.findByLogin("Non existing login"))
+                .isInstanceOf(UserNotFoundException.class)
+                .hasMessage("Пользователь не найден");
+    }
+
+    @Test
+    void getProfileTest() {
+        final UserDto userDTO = userService.saveUser(TestDataProvider.buildAlex());
+        usersId.add(userDTO.getUserId());
+
+        final UserProfileDto userProfileDto = userService.getProfile(userDTO.getLogin());
+        checkUserDTOAndUserProfileDTO(userDTO, userProfileDto);
+    }
+
+    @Test
+    void findAllIsEmptyListTest() {
+        assertThat(userService.findAllUsers(0)).isEmpty();
+    }
+
+    @Test
+    void findAllTest() {
+        final List<UserDto> users = TestDataProvider.buildUsers().stream()
+                .map(u -> userService.saveUser(u))
+                .collect(Collectors.toList());
+        users.forEach(u -> usersId.add(u.getUserId()));
+
+        final var foundUsers = userService.findAllUsers(0);
```

Comment:
Admin functions have already been tested in AdminTest

Reply:
But these are different methods in project logic, maybe I missed something

I think it is better to test them separately or we need to reuse the findAll() from the UserService in AdminService